controllers/authController.js

```diff
@@ -0,0 +1,106 @@
+const bcrypt = require("bcrypt");
+const jwt = require("jsonwebtoken");
+
+const { SECRET_KEY } = process.env;
+
+const { Users } = require("../models/Users");
+const { HttpError } = require("../utils/HttpError");
+const { controllerWrapper } = require("../utils/controllerWrapper");
+
+const register = async (req, res) => {
+  const { email, password } = req.body;
+  const user = await Users.findOne({ email });
+
+  if (user) {
+    throw new HttpError(409, "Email in use");
+  }
+
+  const hashedPassword = await bcrypt.hash(password, 10);
+
+  const newUser = await Users.create({ ...req.body, password: hashedPassword });
+
+  res.status(201).json({
+    user: {
+      email: newUser.email,
+      subscription: newUser.subscription,
+    },
+  });
+};
+
+const login = async (req, res) => {
+  const { email, password } = req.body;
+  const user = await Users.findOne({ email });
+
+  if (!user) {
+    throw new HttpError(401, "Email or password is wrong");
+  }
+
+  const passwordCompared = await bcrypt.compare(password, user.password);
```

Comment:
```suggestion
  const isPasswordCorrect= await bcrypt.compare(password, user.password);
```
 it would be a better name